### PR TITLE
Add debiased distribution compression source code

### PR DESCRIPTION
**Pull request by lingxiaoli94**

### CODE REVIEW

**Comment by lingxiaoli94** [comment]
@microsoft-github-policy-service agree [company="MIT"]

**Comment by lingxiaoli94** [comment]
@microsoft-github-policy-service agree

**Review by lmackey — APPROVED** [review]

**Review by lmackey — CHANGES_REQUESTED** [review]
I'd recommend using a persistent data repository like https://dataverse.harvard.edu/ for hosting this data.

File: goodpoints/jax/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+This package contains algorithms from Debiased Distribution Compression by Li et al. 2024 implemented in JAX.
+
+## Debiased distribution compression algorithms
+The `dtc.py` module contains the meta debiased-then-compress algorithm for debiased compression along with 6 instantiations corresponding to the 6 algorithms in the paper.
```
Comment (lmackey) [inline goodpoints/jax/README.md]: debias-then-compress

File: goodpoints/jax/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+For evaluation, `mmd.py` provides a function to compute the MMD between two sets of weighted points, using tiling to speed up the computation while reducing the memory footprint.
+
+
+The `kernel/` package includes modules to build kernels with automatic kernel parameter tuning. As we often deal with points with additional per-point attributes (such as the score), `sliceable_points.py` provides a JAX data structure to store points with additional attributes with helper functions, which is then input to the kernel.
+
+The `serial/` package provides utilities for serialization and caching of intermediate outputs.
+
+## Dependencies
+The main dependency is JAX whose install instructions can be found at [here](https://jax.readthedocs.io/en/latest/quickstart.html).
```
Comment (lmackey) [inline goodpoints/jax/README.md]: The main dependency is JAX; the install instructions can be found

File: README.md [hunk 3]
```diff
@@ -7,12 +7,15 @@ GoodPoints is a collection of tools for compressing a distribution more effectiv
...
 - **Compress++** reduces the runtime of generic thinning algorithms with minimal loss in accuracy
 - **Compress Then Test** accelerates kernel two-sample testing using high-fidelity compression
+- **Debiased Distribution Compression** provides a suite of JAX-accelerated methods for debiasing and compressing input points, supporting equal-weighted, simplex-weighted, and constant-preserving coresets
 
 ## Installation
 To install the `goodpoints` package, use the following pip command:
 ```
 pip install goodpoints
 ```
+For **Debiased Distribution Compression**, JAX and additional independencies are required. See [goodpoints/jax](goodpoints/jax) for more details.
```
Comment (lmackey) [inline README.md]: dependencies

File: README.md [hunk 4]
```diff
@@ -7,12 +7,15 @@ GoodPoints is a collection of tools for compressing a distribution more effectiv
 - Given an initial summary of n input points, **kernel thinning** returns s << n output points with comparable integration error across a reproducing kernel Hilbert space
 - **Compress++** reduces the runtime of generic thinning algorithms with minimal loss in accuracy
 - **Compress Then Test** accelerates kernel two-sample testing using high-fidelity compression
+- **Debiased Distribution Compression** provides a suite of JAX-accelerated methods for debiasing and compressing input points, supporting equal-weighted, simplex-weighted, and constant-preserving coresets
```
Comment (lmackey) [inline README.md]: I would remove the Debiased Distribution Compression bullet and instead add the following bullet:
- **Stein Kernel Thinning**, **Stein Recombination**, and **Stein Cholesky** deliver unweighted,  simplex-weighted, and constant-preserving coresets that reduce biases due to off-target sampling, tempering, or burn-in

File: README.md [hunk 5]
```diff
@@ -164,6 +175,17 @@ by executing `examples/kt/run_kt_experiment.ipynb` with appropriate parameters.
 ```
 See [examples/mmd_test/README.md](examples/mmd_test/README.md).
 
+#### [Debiased Distribution Compression](https://arxiv.org/pdf/2404.12290)
+```
+@article{li2024debiased,
```
Comment (lmackey) [inline README.md]: InProceedings{li2024debiased,
  title = 	 {Debiased Distribution Compression},
  author =       {Li, Lingxiao and Dwivedi, Raaz and Mackey, Lester},,
  booktitle = 	 {Proceedings of the 41st International Conference on Machine Learning},
  year = 	 {2024},
  volume = 	 {203},
  series = 	 {Proceedings of Machine Learning Research},
  month = 	 {21--27 Jul},
  publisher =    {PMLR},
}

**Review by lmackey — CHANGES_REQUESTED** [review]

**Comment by lingxiaoli94** [comment]
I have addressed the comments and added the link to Harvard dataverse at [here](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/NGEM9H).

**Review by lmackey — APPROVED** [review]

**Comment by lmackey** [comment]
Adding debiased distribution compression code

**Pull request merged**